pkg/buildplan/ingredient.go

```diff
@@ -58,7 +59,8 @@ func (i Ingredients) ToNameMap() IngredientNameMap {
 }
 
 func (i *Ingredient) RuntimeDependencies(recursive bool) Ingredients {
-	return i.runtimeDependencies(recursive, make(map[strfmt.UUID]struct{}))
+	dependencies := i.runtimeDependencies(recursive, make(map[strfmt.UUID]struct{}))
+	return sliceutils.UniqueByProperty(dependencies, func(i *Ingredient) any { return i.IngredientID })
```

Comment:
This feels like it's working around a bug. `runtimeDependencies` shouldn't give duplicate results, if it does then we need to fix that.

I think that's what this line is meant to prevent:

https://github.com/ActiveState/cli/blob/master/pkg/buildplan/ingredient.go#L66-L66

Reply:
Hmm ok, thanks for checking!